Chunks/ChunkManagement/ChunkWorkers/NonGenerativeChunkWorker.cs

```diff
@@ -80,11 +80,32 @@ private void RunJob()
     {
         try
         {
-            while (_chunksToUpdate.TryTake(out var chunk, Timeout.Infinite, _cancellationTokenSource.Token))
+            while (_modificationsToPerform.TryTake(out var modification, Timeout.Infinite, _cancellationTokenSource.Token))
             {
-                chunk.Mesh.Vertices = _meshGenerator.GetMesh(chunk.Position, new ChunkData { SphereId = chunk.Sphere, Voxels = chunk.Voxels });
+                var modificationType = modification.ModificationType;
+                ref var location = ref modification.Location;
```

Comment:
why ref

Reply:
micro-optimization, 3 * sizeof(float)= 12 > sizeof(pointer) = 8

Reply:
I hate it. very small optimization and if someone wants to hange this for whatever reason in the future it will become a bug. Things are passed by value by default for a reason

Reply:
I don't know if it's very error prone, if someone changes it, it'll just change the Location field in a local `modification` variable, and we already have tons of those `ref`var`s in other places

Reply:
`ref` or without, the CIL is the same. Removed it.